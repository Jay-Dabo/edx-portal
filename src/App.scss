@import "~bootstrap/scss/bootstrap";
@import "~@edx/edx-bootstrap/sass/edx/theme";
