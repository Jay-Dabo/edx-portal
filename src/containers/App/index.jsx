import React from 'react';
import { Helmet } from 'react-helmet';
import { Switch, Route } from 'react-router-dom';

import HomePage from '../HomePage';
import CoursewareContainerPage from '../CoursewareContainerPage';
import TermsOfServicePage from '../TermsOfServicePage';
import NotFoundPage from '../NotFoundPage';
import Header from '../../components/Header';
import Footer from '../../components/Footer';

import './App.scss';

export default function App() {
  return (
    <div>
      <Helmet
        titleTemplate="%s - edX Portal"
        defaultTitle="edX Portal"
      />
      <Header />
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/course" component={CoursewareContainerPage} />
        <Route path="/terms-of-service" component={TermsOfServicePage} />
        <Route path="" component={NotFoundPage} />
      </Switch>
      <Footer />
    </div>
  );
}
