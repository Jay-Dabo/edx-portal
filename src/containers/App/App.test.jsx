import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import App from './index';

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create((
        <MemoryRouter>
          <App />
        </MemoryRouter>))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
