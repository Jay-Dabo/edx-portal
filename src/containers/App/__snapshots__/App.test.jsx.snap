// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <header>
    <nav
      className="navbar"
    >
      <div
        className="container"
      >
        <div
          className="navbar-header"
        >
          <a
            className="navbar-brand"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="Enterprise logo"
              className=""
              src="test-file-stub"
            />
          </a>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="container"
  >
    <h1>
      Rockin' the Home Page!
    </h1>
    <p>
      Blah blah blah blah
    </p>
    <a
      href="/admin"
      onClick={[Function]}
    >
      Admin Dashboard
    </a>
  </div>
  <footer
    className="pb-4"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-between align-items-center"
      >
        <div
          className="col-xs-12 col-md-4 logo-links"
        >
          <a
            className="logo"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="edX logo"
              className=""
              src="test-file-stub"
            />
          </a>
          <a
            className="logo"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="Enterprise logo"
              className=""
              src="test-file-stub"
            />
          </a>
        </div>
        <div
          className="col-xs-12 col-md footer-links"
        >
          <nav>
            <ul
              className="nav justify-content-end"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="https://www.edx.org/edx-terms-service"
                >
                  Terms of Service
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="https://www.edx.org/edx-privacy-policy"
                >
                  Privacy Policy
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/faq"
                  onClick={[Function]}
                >
                  FAQ & Support
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
